pkg/koordlet/resmanager/cpu_evict.go

```diff
@@ -306,3 +317,59 @@ func isSatisfactionConfigValid(thresholdConfig *slov1alpha1.ResourceThresholdStr
 	}
 	return true
 }
+
+func getBECPUMetric(resouceAllocation metriccache.MetricPropertyValue, querier metriccache.Querier, aggregateType metriccache.AggregationType) (float64, int64) {
+	var properties map[metriccache.MetricProperty]string
+
+	switch resouceAllocation {
+	case metriccache.BEResouceAllocationUsage:
+		properties = metriccache.MetricPropertiesFunc.NodeBE(string(metriccache.BEResourceCPU), string(metriccache.BEResouceAllocationUsage))
+	case metriccache.BEResouceAllocationRequest:
+		properties = metriccache.MetricPropertiesFunc.NodeBE(string(metriccache.BEResourceCPU), string(metriccache.BEResouceAllocationRequest))
+	case metriccache.BEResouceAllocationRealLimit:
+		properties = metriccache.MetricPropertiesFunc.NodeBE(string(metriccache.BEResourceCPU), string(metriccache.BEResouceAllocationRealLimit))
+	default:
+		properties = map[metriccache.MetricProperty]string{}
+	}
+
+	result, err := doQuery(querier, metriccache.NodeBEMetric, properties)
+	if err != nil {
+		klog.Warningf("cpuEvict by ResourceSatisfaction skipped, %s queryResult error: %v", resouceAllocation, err)
+		return 0.0, 0
+	}
+	value, err := result.Value(aggregateType)
+	if err != nil {
+		klog.Warningf("cpuEvict by ResourceSatisfaction skipped, queryResult %s error: %v", aggregateType, err)
+		return 0.0, 0
+	}
+	count := result.Count()
+
+	return value, int64(count)
+
+}
+
+func doQuery(querier metriccache.Querier, resource metriccache.MetricResource, properties map[metriccache.MetricProperty]string) (metriccache.AggregateResult, error) {
```

Comment:
This can be a common function for resmanager. Perhaps it should be moved into `resmanager/resmanager.go`.